### ISSUE TITLE
Theme is changing when opening a Dropdown

### ISSUE THREAD

**Issue by origami-z** [jpmorganchase/salt-ds#1650]
### Package name(s)

Core v1.7.0 (@salt-ds/core), Lab v1.0.0-alpha.6 (@salt-ds/lab)

### Description

When using below versions of salt packages, Salt provider's `mode` is changed upon opening and closing the dropdown.

```
    "@salt-ds/core": "1.7.0",
    "@salt-ds/data-grid": "1.0.3",
    "@salt-ds/icons": "1.3.0",
    "@salt-ds/lab": "1.0.0-alpha.6",
    "@salt-ds/theme": "1.4.0",
```

### Steps to reproduce

When click the dropdown with below code, mode (HTML `data-mode`) is changed to light, then when close dropdown is closed, mode becomes `undefined` and `salt-theme` classes are also taken away.

Stackblitz: https://stackblitz.com/edit/salt-template-drncvk?file=App.tsx

```
    <SaltProvider mode="dark">
      <StackLayout>
        <Dropdown source={shortColorData} defaultSelected={shortColorData[0]} />
        <Button><ThumbsUpIcon /></Button>
      </StackLayout>
    </SaltProvider>
```

### Expected behavior

Mode should stay as it is

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

**Comment by origami-z** [comment]
This may be related to #1325, where multiple version of core packages are installed

**Comment by origami-z** [comment]
Confirmed this is fixed by #1647
